dashboard/app/models/teacher_feedback.rb

```diff
@@ -57,23 +60,45 @@ def get_script_level
     script_level
   end
 
-  def self.get_student_level_feedback(student_id, level_id, teacher_id, script_id)
+  def self.get_student_level_feedback_by_teacher(student_id, level_id, script_id, teacher_id)
     where(
       student_id: student_id,
       level_id: level_id,
-      teacher_id: teacher_id,
-      script_id: script_id
+      script_id: script_id,
+      teacher_id: teacher_id
     ).latest
   end
 
-  def self.get_all_feedback_for_section(student_ids, level_ids, teacher_id)
-    find(
-      where(
-        student_id: student_ids,
-        level_id: level_ids,
-        teacher_id: teacher_id
-      ).group([:student_id, :level_id]).pluck('MAX(teacher_feedbacks.id)')
-    )
+  def self.get_all_student_level_feedback(student_id, level_id, script_id)
+    where(
+      student_id: params.require(:student_id),
+      level_id: params.require(:level_id),
+      script_id: params.require(:script_id)
```

Comment:
I think you want to use the passed in `student_id`, `level_id`, and `script_id`, right?